prov/util/src/util_shm.c

```diff
@@ -525,8 +543,11 @@ void smr_map_del(struct smr_map *map, int64_t id)
 	pthread_mutex_unlock(&ep_list_lock);
 
 	ofi_spin_lock(&map->lock);
-	if (!entry)
+	if (!entry) {
 		munmap(map->peers[id].region, map->peers[id].region->total_size);
+		if (map->caps & FI_HMEM)
+			(void) ofi_hmem_host_unregister(map->peers[id].region);
+	}
```

Comment:
host unregister should be called before munmap.

Reply:
Updated